grantnav/frontend/views.py

```diff
@@ -491,6 +491,185 @@ def add_advanced_search_information_in_context(context):
     return context
 
 
+def term_facet_from_parameters(request, json_query, field_name, param_name, bool_index, field, is_json=False):
+    new_filter = []
+    for value in request.GET.getlist(param_name):
+        new_filter.append({"term": {field_name: value}})
+
+    json_query["query"]["bool"]["filter"][bool_index]["bool"]["should"] = new_filter
+
+
+def amount_facet_from_parameters(request, json_query):
+    new_filter = []
+    for value in request.GET.getlist('amountAwarded'):
+        for item in FIXED_AMOUNT_RANGES:
+            if value == str(item['from']):
+                new_range = {"gte": item["from"]}
+                to_ = item.get("to")
+                if to_:
+                    new_range["lt"] = to_
+                new_filter.append({"range": {"amountAwarded": new_range}})
+
+    json_query["query"]["bool"]["filter"][2]["bool"]["should"] = new_filter
+
+
+def date_facet_from_parameters(request, json_query):
+    new_filter = []
+    for value in request.GET.getlist("awardDate"):
+        new_filter.append(
+            {"range": {
+                "awardDate": {"format": "year",
+                                  "gte": value + "||/y",
+                                  "lte": value + "||/y"}
+            }
+            }
+        )
+
+    json_query["query"]["bool"]["filter"][4]["bool"]["should"] = new_filter
+
+
+def term_facet_size_from_parameters(request, json_query):
+    try:
+        aggs = json_query["aggs"]
+    except KeyError:
+        aggs = BASIC_QUERY['aggs']
+
+    for agg_name, agg in aggs.items():
+        if "terms" not in agg:
+            continue
+        more = request.GET.get(agg_name+'More')
+        if more:
+            agg["terms"]["size"] = 50
+        else:
+            agg["terms"]["size"] = 3
+
+def non_term_facet_size_from_parameters(request, json_query, agg_name):
+
+    more = request.GET.get(agg_name+'More')
+    if more:
+        json_query['extra_context'][agg_name + '_facet_size'] = 50
+    else:
+        json_query['extra_context'][agg_name + '_facet_size'] = 3
+
+
+def create_json_query_from_parameters(request):
+    json_query = copy.deepcopy(BASIC_QUERY)
+    json_query["query"]["bool"]["must"]["query_string"]["query"] = request.GET.get('query', '*')
+    json_query["query"]["bool"]["must"]["query_string"]["default_field"] = request.GET.get('default_field', '*')
+
+    sort_order = request.GET.get('sort', '').split()
+    if sort_order and len(sort_order) == 2:
+        sort = {sort_order[0]: {"order": sort_order[1]}}
+        json_query["sort"] = sort
+
+    #page?
```

Comment:
left in by accident perhaps or the start of a comment?

Reply:
yep accident was not sure if we needed to do something with page here.